pkg/commands/operation/operation.go

```
@@ -78,41 +80,39 @@ func DownloadVEXRepositories(ctx context.Context, opts flag.Options) error {
 }
 
 // InitBuiltinChecks downloads the built-in policies and loads them
-func InitBuiltinChecks(ctx context.Context, cacheDir string, quiet, skipUpdate bool, checkBundleRepository string, registryOpts ftypes.RegistryOptions) ([]string, error) {
+func InitBuiltinChecks(ctx context.Context, client *policy.Client, skipUpdate bool, registryOpts ftypes.RegistryOptions) (string, error) {
 	mu.Lock()
 	defer mu.Unlock()
+	var err error
 
-	client, err := policy.NewClient(cacheDir, quiet, checkBundleRepository)
-	if err != nil {
-		return nil, xerrors.Errorf("check client error: %w", err)
-	}
+	if skipUpdate {
+		log.Info("No downloadable checks were loaded as --skip-check-update is enabled, loading from existing cache...")
 
-	needsUpdate := false
-	if !skipUpdate {
-		needsUpdate, err = client.NeedsUpdate(ctx, registryOpts)
+		path := client.LoadBuiltinChecks()
+		_, _, err := misconf.CheckPathExists(path)
 		if err != nil {
-			return nil, xerrors.Errorf("unable to check if built-in policies need to be updated: %w", err)
+			msg := fmt.Sprintf("Failed to load existing cache, err: %s falling back to embedded checks...", err.Error())
+			log.Error(msg)
+			return "", xerrors.New(msg)
```

I think it's better to just return an error and log it on the calling side